src/algo/call_stack_keeper.cc

```diff
@@ -29,8 +30,9 @@ bool CallStackKeeper::recursion_limit_reached() const
 
 void CallStackKeeper::recurse(const std::function<void()> &action)
 {
+    const std::string recursion_error = std::string("Recursion limit reached");
```

Comment:
Consider using `const auto recursion_error = std::string(...)` or `const std::string recursion_error = ...`. (Do we have `"..."s` syntax in C++14?)